maitake/src/wait/cell.rs

```diff
@@ -189,6 +229,32 @@ impl fmt::Debug for WaitCell {
     }
 }
 
+// === impl Wait ===
+
+impl Future for Wait<'_> {
+    type Output = Result<(), Error>;
+
+    fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        if !self.registered {
+            match self.cell.register_wait(cx.waker()) {
+                Ok(_) => {
+                    self.registered = true;
+                    Poll::Pending
+                }
+                Err(Error::Busy) => {
+                    // Cell was busy, all we can do is try again later
+                    cx.waker().wake_by_ref();
+                    Poll::Pending
+                }
+                Err(Error::Closed) => Poll::Ready(Err(Error::Closed)),
+            }
+        } else {
+            // We made it to "once", and got polled again, We must be ready!
+            Poll::Ready(Ok(()))
+        }
+    }
```

Comment:
very tiny style nit, take it or leave it: i think it would be a little bit simpler to write this instead:
```suggestion
        if self.registered {
            // We made it to "once", and got polled again, We must be ready!
            return Poll::Ready(Ok(()));
        }
        
        match self.cell.register_wait(cx.waker()) {
            Ok(_) => {
                self.registered = true;
                Poll::Pending
            }
            Err(Error::Busy) => {
                // Cell was busy, all we can do is try again later
                cx.waker().wake_by_ref();
                Poll::Pending
            }
            Err(Error::Closed) => Poll::Ready(Err(Error::Closed)),
        }
    }
```